pinc/RandomRule.inc

```diff
@@ -58,14 +57,14 @@ class RandomRule
             WHERE
                 document = '%s' AND
                 anchor = '%s' AND
-                langcode = '%s'
-        ", mysqli_real_escape_string(DPDatabase::get_connection(), $document),
-            mysqli_real_escape_string(DPDatabase::get_connection(), $anchor),
-            mysqli_real_escape_string(DPDatabase::get_connection(), $langcode));
+                langcode = '%s'",
+            DPDatabase::escape($document),
+            DPDatabase::escape($anchor),
+            DPDatabase::escape($langcode)
+        );
 
-        $result = mysqli_query(DPDatabase::get_connection(), $sql);
-        if(!$result)
-        {
+        $result = DPDatabase::query($sql, FALSE);
```

Comment:
same here, we can remove the false and the throw below probably I would imagine.

Reply:
Same as above: UnexpectedValueException vs DBQueryError